### DISCUSSION TITLE
[Boost Asio] How to opt out of OpenSSL support for my specific port?

### DISCUSSION THREAD

**Discussion by pratikpc** [microsoft/vcpkg#17619] [Q&A] [answered]
Let us say I want to disable Asio's OpenSSL support

By default, Boost Asio enables OpenSSL while Standalone Asio does not and it's opt in.

Now for a toy project, there's a high chance I am not using OpenSSL.

But if vcpkg is added as a submodule dependency, then every time I clone, a build process for OpenSSL starts and it takes way too long.

In order to reduce the build times, I wanted to ask if I could somehow create a boost-asio[no-ssl] port in which OpenSSL will not be added as a dependency? And if I want to, how would I go about it?

I have a feeling I would have to edit the vcpkg.json file. Right?

**Comment by autoantwort** [comment, marked as answer]
With #17373 merged it should be now much faster to build OpenSSL. So maybe try the latest master.  

> I have a feeling I would have to edit the vcpkg.json file. Right?

It should be enough to remove the openssl dependency from `vcpkg/ports/boost-asio/vcpkg.json`

**Reply by pratikpc** [reply]
#17373 only modifies Unix based ports not Windows based ports so it wouldn't have an effect, positive or negative for yours truly

> It should be enough to remove the openssl dependency from vcpkg/ports/boost-asio/vcpkg.json

Yes but for it to work, I would have to either fork VCPKG and link to my own as a submodule which I don't want to do or patch it up everytime.
I was hoping a port like boost-asio[nossl] or boost-asio[-ssl] would be possible

**Reply by mathisloge** [reply]
You can create a https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md to overwrite the asio port. You don't have to patch it everytime

**Reply by pratikpc** [reply]
@mathisloge any idea how to use it with CMake?

**Reply by mathisloge** [reply]
You can specify default overlay ports with env vars.

See https://www.github.com/microsoft/vcpkg/tree/master/docs%2Fusers%2Fconfig-environment.md and scroll a bit down

**Reply by pratikpc** [reply]
Awesome
Thanks @mathisloge

**Reply by dg0yt** [reply]
Alternative to environment variables, there are also CMake variables. The documentations is at:
https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md#cmake-integration

**Reply by pratikpc** [reply]
@dg0yt Awesome
Thanks!
I think your approach is a lot more easier to port
Thanks all of you
